tiledb/CMakeLists.txt

```diff
@@ -795,4 +796,32 @@ configure_file(
         @ONLY
 )
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tiledb.pc
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig)
\ No newline at end of file
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+#-----------------------------------------------------------------------------
+# Generate cpp with git commit hash
+find_package(Git)
+if(Git_FOUND)
+  set(TILEDB_GITVER_FILE "${CMAKE_CURRENT_BINARY_DIR}/tiledb_git_sha.h")
+  set(TILEDB_GITVER_TEMP "${CMAKE_CURRENT_BINARY_DIR}/tiledb_git_sha.tmp")
+
+  # always run this target to check for changes.
+  add_custom_target(_GEN_GITVER ALL
+    # write git hash to temp file
+    COMMAND ${CMAKE_COMMAND} -E echo         "#define _tiledb_xstr(s) #s"             > "${TILEDB_GITVER_TEMP}"
```

Comment:
Instead of an echo, which I don't think works on windows, you can use a compiler define for `TILEDB_GIT_HASH`. You can then keep the `tiledb_git_sha.h` in the repo and wrap a default value in `ifndef TILEDB_GIT_HASH`.

Reply:
This is using ‘CMake -E echo’ (CMake builtin command)